### PR TITLE
REF: make Resolution an enum

### PR DESCRIPTION
**Pull request by jbrockmendel**
I find this much clearer than just using an int.  No measured performance impact.

### CODE REVIEW

File: pandas/_libs/tslibs/resolution.pyx [hunk 1]
```diff
@@ -111,41 +148,14 @@ class Resolution:
...
-        'N': None}
-
-    _str_reso_map = {v: k for k, v in _reso_str_map.items()}
-
-    _freq_reso_map = {v: k for k, v in attrname_to_abbrevs.items()}
+    def __lt__(self, other):
+        return self.value < other.value
+
+    def __ge__(self, other):
+        return self.value >= other.value
```
Comment (mroeschke) [inline pandas/_libs/tslibs/resolution.pyx:155]: Curious why this is `>=` and not just `>`

Comment (jbrockmendel) [reply pandas/_libs/tslibs/resolution.pyx:155]: because "ge" is "greater or equal to"

Comment (mroeschke) [reply pandas/_libs/tslibs/resolution.pyx:156]: Ah right 🤦

**Comment by jreback** [comment]
lgtm if you can run some basic perf would be great

**Comment by jbrockmendel** [comment]
> if you can run some basic perf would be great

mentioned in the OP, asv didnt show measurable change

**Review by mroeschke — APPROVED** [review]

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -3598,9 +3598,10 @@ cpdef to_offset(freq):
                 if not stride:
                     stride = 1
 
-                from .resolution import Resolution  # TODO: avoid runtime import
+                # TODO: avoid runtime import
+                from .resolution import Resolution, reso_str_bump_map
```
Comment (jreback) [inline pandas/_libs/tslibs/offsets.pyx:3602]: can you do an absolute import

**Review by jreback — CHANGES_REQUESTED** [review]
small comment, ping on green.

File: pandas/_libs/tslibs/offsets.pyx (hunk 2 above)
Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:3602]: sure, follow-up?  this is a blocker for stuff im excited about

Comment (jbrockmendel) [reply pandas/_libs/tslibs/offsets.pyx:3602]: (and actually #34486 will get rid of this import)

**Review by jreback — APPROVED** [review]

**Pull request merged**